src/app/public/src/modules/clipboard/clipboard.directive.ts

```diff
@@ -0,0 +1,80 @@
+// adapted from https://github.com/maxisam/ngx-clipboard/
+import {
+  Directive,
+  EventEmitter,
+  OnDestroy,
+  Output,
+  Renderer2,
+  ElementRef,
+  AfterViewInit,
+  ChangeDetectorRef,
+  Input
+} from '@angular/core';
+
+import { StacheClipboardService } from './clipboard.service';
+
+@Directive({
+  selector: '[stacheClipboard]'
+})
+export class StacheClipboardDirective implements AfterViewInit, OnDestroy {
+  // --------------------------------------------------------------
+  // NOTE: Only supported for Code-Block component as of July 2018
+  // --------------------------------------------------------------
+
+  private defaultTigger = 'clipboardButton';
+
+  @Input() public clipboardTrigger: string = this.defaultTigger;
+  @Input() public copySuccessText: string;
+  @Output() public cbOnSuccess: EventEmitter<any> = new EventEmitter<any>();
```

Comment:
The word "on" is discouraged by Angular style guide: https://angular.io/guide/styleguide#dont-prefix-output-properties.

We generally name events that happen multiple times in the present tense, and events that occur only once in the past tense.

Also, are these outputs even needed at this point? I tend to "under" release features to avoid complexity, and only add features when they're directly asked for. Since this directive is internal-only, we can avoid overcomplicating the API until it's released to the public.

Reply:
Definitely. Some of these pieces are left overs from the implementation we are pulling the directive and directive service from that we should remove if we don't need/want them.